projects/cashmere/src/lib/modal/modal.spec.ts

```diff
@@ -30,24 +37,164 @@ describe('hc-modal', () => {
         windowComponent = windowFixture.componentInstance;
         windowFixture.detectChanges();
 
-        overlayFixture = TestBed.createComponent(ModalOverlayComponent);
-        overlayComponent = overlayFixture.componentInstance;
-        overlayFixture.detectChanges();
+        headerFixture = TestBed.createComponent(ModalHeaderComponent);
+        headerComponent = headerFixture.componentInstance;
+        headerFixture.detectChanges();
+
+        footerFixture = TestBed.createComponent(ModalFooterComponent);
+        footerComponent = footerFixture.componentInstance;
+        footerFixture.detectChanges();
     });
 
     it('should create a window component', () => {
         expect(windowComponent).toBeTruthy();
     });
 
-    it('should create an overlay component', () => {
-        expect(overlayComponent).toBeTruthy();
+    it('should create an header component', () => {
+        expect(headerComponent).toBeTruthy();
+    });
+
+    it('should create an footer component', () => {
+        expect(footerComponent).toBeTruthy();
+    });
+
+    it('modal window should to be draggable', () => {
+        windowComponent._isDraggable = true;
+        windowFixture.detectChanges();
+        expect(windowComponent._isDraggable).toBeTruthy();
+    });
+
+    it('should the overlay click be disable', () => {
+        windowComponent._ignoreOverlayClick = true;
+        windowFixture.detectChanges();
+        expect(windowComponent._ignoreOverlayClick).toBeTruthy();
     });
 
     it('should change size depending on options', () => {
         let div: DebugElement = windowFixture.debugElement.query(By.css('.hc-modal'));
         expect(div.properties['className']).toContain('hc-modal-auto');
+        // Small
```

Comment:
Each of these size tests should be in their own test.  You can surround the tests with a general description by using 'describe', like this:

```
describe('should change size', () => {
     it('when window is small'), () => {
          ...
     });

     it('when window is medium'), () => {
          ...
     });
});
```

Reply:
Great suggestions @Techgeekster - thank you!